srcdeps-maven-plugin/src/main/java/org/srcdeps/mvn/plugin/ExtensionsXmlUtils.java

```diff
@@ -0,0 +1,164 @@
+ * limitations under the License.
+ */
+package org.srcdeps.mvn.plugin;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.w3c.dom.Text;
+
+/**
+ * Static methods to modify {@code extensions.xml} DOM.
+ *
+ * @author <a href="https://github.com/ppalaga">Peter Palaga</a>
+ */
+public class ExtensionsXmlUtils {
+
+    public enum ExtensionsXmlAttribute {
+        artifactId, groupId, version
+    }
+
+    public enum ExtensionsXmlElement {
+        artifactId, extension, extensions, groupId, version
+    }
+
+    public enum SrcdepsExtension {
+        localRepo("org.srcdeps.mvn", "srcdeps-maven-local-repository"), //
+        enforcer("org.srcdeps.mvn", "srcdeps-maven-enforcer") //
+        ;
+
+        private final String artifactId;
+        private final String groupId;
+
+        SrcdepsExtension(String groupId, String artifactId) {
+            this.groupId = groupId;
+            this.artifactId = artifactId;
+        }
+
+        public boolean matches(String groupId, String artifactId) {
+            return this.groupId.equals(groupId) && this.artifactId.equals(artifactId);
+        }
+    }
+
+    private static String getFirstChildTextValue(Element parent, String childName) {
+        NodeList children = parent.getChildNodes();
+        for (int i = 0; i < children.getLength(); i++) {
+            Node child = children.item(i);
+            if (child instanceof Element) {
+                Element elem = (Element) child;
+                if (childName.equals(elem.getNodeName())) {
+                    return getOrCreateFirstTextChild(elem).getTextContent();
+                }
+            }
+        }
+        return null;
+    }
+
+    private static Text getOrCreateFirstTextChild(Element node) {
+        NodeList children = node.getChildNodes();
+        for (int i = 0; i < children.getLength(); i++) {
+            Node child = children.item(i);
+            if (child instanceof Text) {
+                return (Text) child;
+            }
+        }
+        Text result = node.getOwnerDocument().createTextNode("");
+        node.appendChild(result);
+        return result;
+    }
+
+    private static void setTextValue(Element parent, String childName, String childValue) {
+        Element elem = parent.getOwnerDocument().createElement(childName);
+        parent.appendChild(elem);
+        Text textNode = parent.getOwnerDocument().createTextNode(childValue);
+        elem.appendChild(textNode);
+    }
+
+    /**
+     * Adds or upgrades to the given {@code newVersion} the srcdeps extensions listed in {@link SrcdepsExtension} in the
+     * given DOM {@code document}.
+     *
+     * @param doc
+     *            the DOM of {@code extensions.xml} to modify, may be an empty DOM
+     * @param newVersion
+     *            the srcdeps version to add or upgrade to
+     */
+    public static void upgradeOrAddSrcdeps(Document doc, String newVersion) {
```

Comment:
Where is this used, in which context? Should I expect stuff on my repo to change if this code runs? For instance, will it change my YAML configuration file?